src/libcore/slice/mod.rs

```diff
@@ -5773,6 +5773,25 @@ unsafe impl<'a, T> TrustedRandomAccess for RChunksExactMut<'a, T> {
 /// assert_eq!(slice[0], 42);
 /// ```
 ///
+/// ## Counterexample
+///
+/// Objects from different allocations may end up contiguous in memory, so the following function is unsound:
```

Comment:
I think it would be good, here or in the comments in the code, to explicitly use the same terminology as the safety clause above:

> The entire memory range of this slice must be contained within a single allocated object!